lib/detectors/DBZ/DBZ.h

```diff
@@ -47,6 +47,19 @@ namespace vanguard {
         using Val2Signal = std::map<const llvm::Value *, typename Domain::template Signal<llvm::Value> *>;
         using SignalVec = std::vector<typename Domain::template Signal<llvm::Value> *>;
 
+        /**
+        * Finds all the Binary Operators in a function
+        */
+        class BinaryOpQuery : public InstructionClassQuery<Domain, const llvm::Value *> {
+        public:
+            virtual void visit(const UnknownIns <Domain> &v) {
```

Comment:
```suggestion
            virtual void visit(const BinaryOpIns <Domain> &v) {
```

There are several classes that wrap instructions used depending on the subjacent llvm instruction. For the instructions you want I think you need to use `BinaryOpIns`